File: BruteForce/BOJ_2231_분해합.py
# BOJ 2231번
# 분해 합
# Bronze 2

N = input() # N - String

def f(N):
    for M in range(1, int(N)): # M - int
    # for M in range(int(N)-9*len(N), int(N)): # 이렇게 하면 안됨. 1자릿수일 때 고려를 못함.

        # 자릿수 합
        sum_loc = M + sum(map(int, str(M)))
        
        if sum_loc == int(N):
            return M # M - int
    
    return 0

print(f(N))

File: BruteForce/BOJ_2231_분해합.cpp
// BOJ 2231번
// 분해합
// Bronze 2

#include <iostream>
using namespace std;


int decomposition(int N) {
    for (int M = 1; M < N; M++) {
        int sum = M;
        int temp = M;

        while (temp > 0) {
            sum += temp%10;
            temp /= 10;
        }

        if (sum == N)
            return M;  
    }
    return 0;
}

int main() {
    int N;
    cin >> N;

    cout << decomposition(N);

    return 0;
}